Update django-friends to be compatible with django 2.0

flowerncsu opened the pull request:
Updates to make django-friends compatible with Django 2.0.

I'm not sure why the changes to the readme are indicated as being part of this; I don't believe I made those changes. I'm happy to re-submit with the readme changes pulled out if that's preferred.

KatherineMichel commented:
It looks like you began working on this branch before you submitted some updates via your documentation branch. It looks like this branch is not up to date with those changes.

flowerncsu commented:
The docs conflict is resolved. Travis failed initially because of a few things that were legitimate issues (which I fixed) but also because I created a migration file, which doesn't adhere to PEP 8. I updated the travis spec to exclude migrations from the style check, because my preference is always to have the migrations in libraries, so that the user doesn't have to use `makemigrations` before using the library (ie, I try to be consistent with the django libraries themselves. `auth`, for example, has its migrations pre-created). If you prefer to not have migrations pre-created, that can be changed, of course.

KatherineMichel commented:
@flowerncsu Thank you for working on this! We've actually begun to convert apps from Travis CI to Circle CI. We also use tox/detox to make sure 2.0 changes are compatible. There is some info here if you are interested in making the conversion. https://github.com/pinax/pinax/wiki/Converting-to-CircleCi-and-Codecov

KatherineMichel commented:
@flowerncsu Thank you for doing more work on this! Were you able to get detox to pass?

flowerncsu commented:
No, but I'm not at all sure I have tox installed/running correctly locally. I haven't worked much with tox, and none at all for at least a year, so I'm pretty rusty with it. Is there an instruction doc for running tox for pinax like the one you pointed me to for doing the conversion?

KatherineMichel commented:
@flowerncsu Here is the one that I used to learn how: [How to Test Against Multiple Python Versions in Parallel](http://blog.pinaxproject.com/2015/12/08/how-test-against-multiple-python-versions-parallel/). It's from 2015 and a bit out of date (Travis CI is no longer used, for instance and the tox.ini will not be the same as the one in the more recent conversion docs. Btw, if you are using pipenv, you can run tox using ```pipenv run detox```. We also sometimes use ```isort --recursive pinax -sp tox.ini``` to sort imports.

flowerncsu commented:
Hm. It seems like tox is set up correctly, but isn't initializing the django settings module? I'm not sure where the problem is here. This is what I get when I run tox locally on this branch: https://pastebin.com/0fxJs9Eu

KatherineMichel commented:
Chatted about this in Slack. More progress will be attempted at a later time, unless anyone else has input.
